### PR TITLE
Add AutoMaterializePolicies to toys

### PR DESCRIPTION
**Pull request by johannkm**
Some basic examples to play with
Screen Shot 2023-04-13 at 12 52 30 AM

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13590** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13590" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13590?utm_source=stack-comment).

File: python_modules/dagster-test/dagster_test/toys/auto_materializing/assets.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+from dagster import AssetKey, AutoMaterializePolicy, SourceAsset, asset, repository
+from dagster._core.definitions.freshness_policy import FreshnessPolicy
+
+# EAGER
+
+
+@asset(auto_materialize_policy=AutoMaterializePolicy.lazy())
+def eager_upstream():
```
Comment (sryza) [inline python_modules/dagster-test/dagster_test/toys/auto_materializing/assets.py:8]: I find it a little confusing that this asset with `eager` in the name has a lazy materialization policy.

**Review by sryza — APPROVED** [review]

File: python_modules/dagster-test/dagster_test/toys/auto_materializing/assets.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+@asset(
+    auto_materialize_policy=AutoMaterializePolicy.lazy(),
+    freshness_policy=FreshnessPolicy(maximum_lag_minutes=5),
+)
+def lazy_downstream_4(lazy_downstream_2, lazy_downstream_3):
+    return lazy_downstream_2 + lazy_downstream_3
+
+
+@repository
```
[the remark this answers is not in the document]
Comment (johannkm) [reply python_modules/dagster-test/dagster_test/toys/auto_materializing/assets.py:84]: there's two repos here- moving it to Definitions will mean we'd need to add a second code location

[the remark this answers is not in the document]
Comment (johannkm) [reply python_modules/dagster-test/dagster_test/toys/auto_materializing/assets.py:84]: I'll add a comment about why these are separate repos

[the remark this answers is not in the document]
Comment (sryza) [reply python_modules/dagster-test/dagster_test/toys/auto_materializing/assets.py:84]: Could we split this into two .py files that we import into separate repos? It seems like they're distinct examples, right?

**Comment by johannkm** [comment]
Yep that works

**Pull request merged**